neqo-bin/benches/main.rs

```diff
@@ -0,0 +1,93 @@
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+use std::{path::PathBuf, str::FromStr};
+
+use criterion::{criterion_group, criterion_main, BatchSize, Criterion, Throughput};
+use neqo_bin::{client, server};
+use neqo_common::log;
+use tokio::runtime::Runtime;
+
+struct Benchmark {
+    name: String,
+    requests: Vec<u64>,
+    /// Download resources in series using separate connections.
+    download_in_series: bool,
+    sample_size: Option<usize>,
+}
+
+fn transfer(c: &mut Criterion) {
+    log::init(Some(log::LevelFilter::Off));
+    neqo_crypto::init_db(PathBuf::from_str("../test-fixture/db").unwrap());
+
+    let done_sender = spawn_server();
```

Comment:
As of today, `cargo bench` is pinned to a single core.

https://github.com/mozilla/neqo/blob/bb3ab602481c0ec61597ef76242a3008108d524b/.github/workflows/bench.yml#L55-L59

Thus `neqo-client` and `neqo-server` compete for the same CPU resource in the benchmarks below.

This could explain the significantly lower throughput on the benchmark server compared to my laptop:

Benchmark server:
```
1-conn/1-100mb-resp/client
                        time:   [12.316 s 12.378 s 12.437 s]
                        thrpt:  [8.0404 MiB/s 8.0786 MiB/s 8.1195 MiB/s]
```
https://github.com/mozilla/neqo/actions/runs/8402727182/job/23012699901?pr=1758

My laptop:
```
1-conn/1-100mb-resp/client
                        time:   [407.05 ms 421.78 ms 439.20 ms]
                        thrpt:  [227.69 MiB/s 237.09 MiB/s 245.67 MiB/s]
```

Reply:
@larseggert do you have thoughts on multi-threaded benchmarks?